src/idl_gen_cpp.cpp

```diff
@@ -878,13 +882,19 @@ class CppGenerator : public BaseGenerator {
     return name.substr(0, name.size() - strlen(UnionTypeFieldSuffix()));
   }
 
-  std::string GetUnionElement(const EnumVal &ev, bool wrap, bool actual_type,
-                              bool native_type = false) {
+  std::string GetUnionElement(const EnumVal &ev, bool wrap_namespace,
```

Comment:
This method looks over complicated. This method (its signature) should be simplified.
Calls:
- L1271: GetUnionElement(ev, true, true, false, false, opts_)
- L1335: GetUnionElement(ev, true, true, true, true, opts_)
- L1368: GetUnionElement(ev, true, true, true, true, opts_)
- L1423: GetUnionElement(ev, true, true, false, false, opts_)
- L1479: GetUnionElement(ev, true, true, false, false, opts_)
- L1510: GetUnionElement(ev, true, true, true, true, opts_)
- L1512: GetUnionElement(ev, **false**, true, false, false, opts_)
- L1545: GetUnionElement(ev, true, true, true, true, opts_)
- L1943: GetUnionElement(ev, true, true, false, false, opts_)
- L2239: GetUnionElement(ev, true, true, false, false, opts_)

Summary:
- there is the only place where `wrap_namespace` is `false` (`wrap_native` is also `false` at this place).
- the `actual_type` never set to `false`.
- both  `native_type` and `wrap_native` have the same value at every call point.

I think that `GetUnionElement` could be split into two independent methods, one for `Type` and another for `Name`.
Usually `GetUnionElement` returns `Type` ( `wrap_namespace=true`). Returned type is either a fully qualified type or `String`. The unused `actual_type` did matter for strings only, however, it never has been used.
The only place (L1512) calls it for `Name` with  `wrap_namespace=false`.

Reply:
Mostly done. I got rid of two of the four booleans. I considered splitting GetUnionElement into GetUnion{Type,Name}, but I think they would share enough code to justify keeping it a single method with an additional bool.